app/hotels/src/allHotels/AllHotelsSearchList.js

```diff
@@ -7,73 +7,77 @@ import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
 
 import AllHotelsSearchRow from './AllHotelsSearchRow';
 import type { AllHotelsSearchList_data as AllHotelsSearchListProps } from './__generated__/AllHotelsSearchList_data.graphql';
-import { withHotelsContext, type HotelsContextState } from '../HotelsContext';
+import { HotelsContext, type HotelsContextState } from '../HotelsContext';
 
 type Props = {|
   +data: ?AllHotelsSearchListProps,
-  +setHotelId: (id: string) => void,
   +ListFooterComponent: React.Node,
 |};
 
 type HotelType = {|
   +id: string,
 |};
 
-export class AllHotelsSearchList extends React.Component<Props> {
-  componentDidMount() {
-    const data = this.props.data ?? [];
+const keyExtractor = (item: ?HotelType) => item?.id;
 
-    if (data.length > 0 && data[0].hotelId != null) {
-      this.props.setHotelId(data[0].hotelId);
-    }
-  }
-
-  keyExtractor = (item: ?HotelType) => item?.id;
+const renderItem = ({
+  item,
+  index,
+}: {|
+  +item: ?HotelType,
+  +index: number,
+|}) => (
+  <AllHotelsSearchRow
+    key={item?.id}
+    data={item}
+    testID={index === 0 ? 'firstHotelResult' : ''}
+  />
+);
 
-  renderItem = ({ item, index }: {| +item: ?HotelType, +index: number |}) => (
-    <AllHotelsSearchRow
-      key={item?.id}
-      data={item}
-      testID={index === 0 ? 'firstHotelResult' : ''}
-    />
-  );
+function getFirst<T>(list: $ReadOnlyArray<T>): ?T {
```

Comment:
I have two questions about this file:
1. is there a reason to use `function` instead of an arrow function?
2. why did you extract this functions from the body of the component? Does that mean they will "never" be garbage collected since they're not part of the component?

Reply:
1. Not really
2. 

> why did you extract this functions from the body of the component
They don't use any props, so they might as well be outside the component. 

> Does that mean they will "never" be garbage collected since they're not part of the component?

🤷‍♂️